qfieldsync/gui/cloud_transfer_dialog.py

```diff
@@ -516,6 +517,36 @@ def build_files_tree(self):
         # ##########
         stack = []
 
+        offline_layers_paths = []
+        if self.cloud_project and self.cloud_project.is_current_qgis_project:
+            project_layers = list(QgsProject.instance().mapLayers().values())
+            for project_layer in project_layers:
+                layer_source = LayerSource(project_layer)
+                if (
+                    layer_source.cloud_action == SyncAction.OFFLINE
+                    and layer_source.filename
+                ):
+                    offline_layers_paths.append(layer_source.filename)
+        elif self.cloud_project and len(self.cloud_project.root_project_files) == 1:
+            project = QgsProject()
+
+            read_flags = QgsProject.ReadFlags()
+            read_flags |= QgsProject.FlagDontResolveLayers
+            read_flags |= QgsProject.FlagDontLoadLayouts
+            if Qgis.versionInt() >= 32600:
+                read_flags |= QgsProject.FlagDontLoad3DViews
+
+            project.read(str(self.cloud_project.root_project_files[0]), read_flags)
+
+            project_layers = list(project.mapLayers().values())
+            for project_layer in project_layers:
+                provider_metadata = QgsProviderRegistry.instance().providerMetadata(
+                    project_layer.providerType()
+                )
+                metadata = provider_metadata.decodeUri(project_layer.source())
+                if metadata.get("path", "") != "":
+                    offline_layers_paths.append(metadata.get("path"))
```

Comment:
```suggestion
        offline_layers_paths = self._get_offline_layers()
```

I think this is quite self-contained to become a separate function and improve readability.